@mqliang the test failures appear related to your change.

**integration.TestAuthModeAlwaysAllow**
```
auth_test.go:474: error in trying to extract resource version: unexpected error, id not found in JSON response: {
	  "kind": "Status",
	  "apiVersion": "v1",
	  "metadata": {},
	  "status": "Failure",
	  "message": "pod a does not have a host assigned",
	  "reason": "BadRequest",
	  "code": 400
	}
auth_test.go:474: error in trying to extract resource version: unexpected error, id not found in JSON response: {
	  "kind": "Status",
	  "apiVersion": "v1",
	  "metadata": {},
	  "status": "Failure",
	  "message": "pod a does not have a host assigned",
	  "reason": "BadRequest",
	  "code": 400
	}
auth_test.go:463: case {POST /api/v1/namespaces/default/events?timeout=60s 
	{
	  "kind": "Event",
	  "apiVersion": "v1",
	  "metadata": {
	    "name": "a"
	  },
	  "involvedObject": {
	    "kind": "Node",
	    "namespace": "default",
	    "name": "a",
	    "apiVersion": "v1"
	  }
	}
	 map[201:true]}
auth_test.go:464: Expected status one of map[201:true], but got 422
auth_test.go:465: Body: {
	  "kind": "Status",
	  "apiVersion": "v1",
	  "metadata": {},
	  "status": "Failure",
	  "message": "Event \"a\" is invalid: involvedObject.namespace: invalid value 'default', Details: namespace is not required for node",
	  "reason": "Invalid",
	  "details": {
	    "name": "a",
	    "kind": "Event",
	    "causes": [
	      {
	        "reason": "FieldValueInvalid",
	        "message": "invalid value 'default', Details: namespace is not required for node",
	        "field": "involvedObject.namespace"
	      }
	    ]
	  },
	  "code": 422
	}
auth_test.go:463: case {PUT /api/v1/namespaces/default/events/a?timeout=60s 
	{
	  "kind": "Event",
	  "apiVersion": "v1",
	  "metadata": {
	    "name": "a",
	"namespace": "default"
	  },
	  "involvedObject": {
	    "kind": "Node",
	    "namespace": "default",
	    "name": "a",
	    "apiVersion": "v1"
	  }
	}
	 map[200:true]}
auth_test.go:464: Expected status one of map[200:true], but got 422
auth_test.go:465: Body: {
	  "kind": "Status",
	  "apiVersion": "v1",
	  "metadata": {},
	  "status": "Failure",
	  "message": "Event \"a\" is invalid: involvedObject.namespace: invalid value 'default', Details: namespace is not required for node",
	  "reason": "Invalid",
	  "details": {
	    "name": "a",
	    "kind": "Event",
	    "causes": [
	      {
	        "reason": "FieldValueInvalid",
	        "message": "invalid value 'default', Details: namespace is not required for node",
	        "field": "involvedObject.namespace"
	      }
	    ]
	  },
	  "code": 422
	}
auth_test.go:463: case {GET /api/v1/namespaces/default/events/a  map[200:true]}
auth_test.go:464: Expected status one of map[200:true], but got 404
auth_test.go:465: Body: {
	  "kind": "Status",
	  "apiVersion": "v1",
	  "metadata": {},
	  "status": "Failure",
	  "message": "events \"a\" not found",
	  "reason": "NotFound",
	  "details": {
	    "name": "a",
	    "kind": "events"
	  },
	  "code": 404
	}
auth_test.go:463: case {DELETE /api/v1/namespaces/default/events/a?timeout=60s  map[200:true]}
auth_test.go:464: Expected status one of map[200:true], but got 404
auth_test.go:465: Body: {
	  "kind": "Status",
	  "apiVersion": "v1",
	  "metadata": {},
	  "status": "Failure",
	  "message": "events \"a\" not found",
	  "reason": "NotFound",
	  "details": {
	    "name": "a",
	    "kind": "events"
	  },
	  "code": 404
	}
auth_test.go:474: error in trying to extract resource version: unexpected error, id not found in JSON response: {
	  "kind": "Status",
	  "apiVersion": "v1",
	  "metadata": {},
	  "status": "Success",
	  "code": 201
	}
auth_test.go:474: error in trying to extract resource version: unexpected error, id not found in JSON response: {
	  "kind": "Status",
	  "apiVersion": "v1",
	  "metadata": {},
	  "status": "Failure",
	  "message": "the server could not find the requested resource",
	  "reason": "NotFound",
	  "details": {},
	  "code": 404
	}
```

**integration.TestAliceNotForbiddenOrUnauthorized**
```
auth_test.go:677: case {POST /api/v1/namespaces/default/events?timeout=60s 
	{
	  "kind": "Event",
	  "apiVersion": "v1",
	  "metadata": {
	    "name": "a"
	  },
	  "involvedObject": {
	    "kind": "Node",
	    "namespace": "default",
	    "name": "a",
	    "apiVersion": "v1"
	  }
	}
	 map[201:true]}
auth_test.go:678: Expected status one of map[201:true], but got 422
auth_test.go:679: Body: {
	  "kind": "Status",
	  "apiVersion": "v1",
	  "metadata": {},
	  "status": "Failure",
	  "message": "Event \"a\" is invalid: involvedObject.namespace: invalid value 'default', Details: namespace is not required for node",
	  "reason": "Invalid",
	  "details": {
	    "name": "a",
	    "kind": "Event",
	    "causes": [
	      {
	        "reason": "FieldValueInvalid",
	        "message": "invalid value 'default', Details: namespace is not required for node",
	        "field": "involvedObject.namespace"
	      }
	    ]
	  },
	  "code": 422
	}
auth_test.go:677: case {PUT /api/v1/namespaces/default/events/a?timeout=60s 
	{
	  "kind": "Event",
	  "apiVersion": "v1",
	  "metadata": {
	    "name": "a",
	"namespace": "default"
	  },
	  "involvedObject": {
	    "kind": "Node",
	    "namespace": "default",
	    "name": "a",
	    "apiVersion": "v1"
	  }
	}
	 map[200:true]}
auth_test.go:678: Expected status one of map[200:true], but got 422
auth_test.go:679: Body: {
	  "kind": "Status",
	  "apiVersion": "v1",
	  "metadata": {},
	  "status": "Failure",
	  "message": "Event \"a\" is invalid: involvedObject.namespace: invalid value 'default', Details: namespace is not required for node",
	  "reason": "Invalid",
	  "details": {
	    "name": "a",
	    "kind": "Event",
	    "causes": [
	      {
	        "reason": "FieldValueInvalid",
	        "message": "invalid value 'default', Details: namespace is not required for node",
	        "field": "involvedObject.namespace"
	      }
	    ]
	  },
	  "code": 422
	}
auth_test.go:677: case {GET /api/v1/namespaces/default/events/a  map[200:true]}
auth_test.go:678: Expected status one of map[200:true], but got 404
auth_test.go:679: Body: {
	  "kind": "Status",
	  "apiVersion": "v1",
	  "metadata": {},
	  "status": "Failure",
	  "message": "events \"a\" not found",
	  "reason": "NotFound",
	  "details": {
	    "name": "a",
	    "kind": "events"
	  },
	  "code": 404
	}
auth_test.go:677: case {DELETE /api/v1/namespaces/default/events/a?timeout=60s  map[200:true]}
auth_test.go:678: Expected status one of map[200:true], but got 404
auth_test.go:679: Body: {
	  "kind": "Status",
	  "apiVersion": "v1",
	  "metadata": {},
	  "status": "Failure",
	  "message": "events \"a\" not found",
	  "reason": "NotFound",
	  "details": {
	    "name": "a",
	    "kind": "events"
	  },
	  "code": 404
	}
```

--- a/pkg/api/validation/events.go
+++ b/pkg/api/validation/events.go
@@ -24,7 +24,11 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) validation.ErrorList {
 	allErrs := validation.ErrorList{}
-	// TODO: There is no namespace required for node.
+	// There is no namespace required for node.
+	if event.InvolvedObject.Kind == "Node" &&
+		event.Namespace != "" {
+		allErrs = append(allErrs, validation.NewInvalidError("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace is not required for node"))
+	}
 	if event.InvolvedObject.Kind != "Node" &&
 		event.Namespace != event.InvolvedObject.Namespace {
 		allErrs = append(allErrs, validation.NewInvalidError("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))

--- a/test/integration/auth_test.go
+++ b/test/integration/auth_test.go
@@ -173,7 +173,7 @@ var aEvent string = `
     "name": "a"%s
   },
   "involvedObject": {
-    "kind": "Node",
+    "kind": "Pod",
     "namespace": "default",
     "name": "a",
     "apiVersion": "v1"

